src/model/model.nit

```diff
@@ -1188,6 +1248,287 @@ abstract class MType
 	end
 end
 
+# A type that represents an operation over a set of types.
+abstract class MTypeSet[E: MType]
+	super MType
+
+	# The module of reference.
+	#
+	# Used to populate `model`, find primitive types and internally build new
+	# types with the same operator.
+	#
+	# SEE: `apply_to`
+	var mmodule: MModule
```

Comment:
OK, I understand how/where it is used but which mmodule is it? or is it any module? Other types are not linked to a module so I honestly don't know.

Not to insist but the only one implementation of  `apply_to` I see in this PR doesn't use it. However, it does a parameter of the same name.